### DISCUSSION TITLE
karmada admin on boarding

### DISCUSSION THREAD

**Discussion by RainbowMango** [karmada-io/karmada#1044] [Ideas]
Many thanks to @prodanlabs for his excellent work on [karmada admin](https://github.com/prodanlabs/kaadm). 
As discussed at the [last community meeting ](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit?pli=1#heading=h.8wspjl3a6x8w), @prodanlabs will contribute the tools to the Karmada community.

We'd like to ask for some advice about where to resettle it.

Alternative 1:
Integrate it to `kubectl-karmada` command by adding a sub-command `install`, e.g.
```
kubectl karmada install --master=xxx.xxx.xxx.xxx
```

Alternative 2:
Introduced another command tool `kaadm`(or `karadm`/`karmadadm`), e.g.
```
kaadm install --master=xxx.xxx.xxx.xxx
```

**Comment by lonelyCZ** [comment]
If there is only a subcommand `install`, I think it is better to integrate it to `kubectl-karmada`, users only download one tool  conveniently.

If considering there will be other subcommands to expand ways of installation in `karadm` in future, it should be introduced another command. 

I prefer `karadm`.

**Comment by RainbowMango** [comment]
Hi @prodanlabs,
How many commands do we need? 
Just put my thoughts here:
- `install` - Install Karmada on a cluster
- `uninstall` - Uninstall Karmada from a cluster
- `upgrade` - Upgrade Karmada
- `init` - Setup Karmada control plane(just like `kubeadm init`)

These commands as well as `join` and `unjoin` could be considered `Administration Commands`.

**Comment by prodanlabs** [comment]
Hi @RainbowMango ,
Same as I thought. 
certainly, @lonelyCZ is also right, users only download one tool conveniently, the original intention of `kaadm` is also for convenience.
I think the two are not in conflict, just like `kubectl` and `kubeadm`.

**Comment by GitHubxsy** [comment]
I prefer the integration into `kubectl-karmada`. 
Karmada users can easily experience excellent features. If `kubeadm` gets bigger in the future, we can consider independent new projects.

**Reply by RainbowMango** [reply]
+1 from me.
@prodanlabs How do you think?
